src/Domain/Bootstrap.php

```diff
@@ -100,7 +100,31 @@ protected function init() {
 	 * @return boolean
 	 */
 	protected function has_core_dependencies() {
-		return class_exists( 'WooCommerce' ) && function_exists( 'register_block_type' );
+		$has_needed_dependencies = class_exists( 'WooCommerce' ) && function_exists( 'register_block_type' );
+		if ( $has_needed_dependencies ) {
+			$plugin_data = \get_file_data(
```

Comment:
I'll double-check on this. The advantage of headers is it's one less thing to update (compared with having a hardcoded string). I don't think the performance hit should be significant because WP only pulls the first 8kb of the files and the parsing is fairly straightforward. I also am only passing in the single array value to check for (which is implemented in the regex pattern).

Reply:
That's fair, but hardcoded values will still perform better, less can go wrong, and it will be easier to read. If we're going to go ahead with automated version changes in the deploy script (as a follow up), it shouldn't matter if there is one more version string to define. We're already doing something similar for `$minimum_wp_version`.

Reply:
Yup agreed, as noted in my feedback on a later comment I'm going to try implementing that.

Reply:
Actually, I just remembered why I took this route. I had to put this in the bootstrap class because otherwise the code change complexity involved in _aborting the loading of Woo Blocks_ is greater. With the WP check, the main file can just return early, but the WC check has to happen (at a minimum) after `plugins_loaded`. So I either have to deactivate the feature plugin when minimum WC version isn't supported, or do some refactoring to support early aborting. All viable solutions but would likely involve more code changes.

Reply:
I just did some rough performance checks on this code using this:

```php
protected function has_core_dependencies() {
		$has_needed_dependencies = class_exists( 'WooCommerce' ) && function_exists( 'register_block_type' );
		$start_time              = hrtime( true );
		if ( $has_needed_dependencies ) {
			$plugin_data = \get_file_data(
				$this->package->get_path( 'woocommerce-gutenberg-products-block.php' ),
				[
					'RequiredWCVersion' => 'WC requires at least',
				]
			);
			if ( isset( $plugin_data['RequiredWCVersion'] ) && version_compare( \WC()->version, $plugin_data['RequiredWCVersion'], '<' ) ) {
				$has_needed_dependencies = false;
				add_action(
					'admin_notices',
					function() {
						if ( should_display_compatibility_notices() ) {
							?>
							<div class="notice notice-error is-dismissible">
								<p><?php esc_html_e( 'WooCommerce Blocks requires a more recent version of WooCommerce and has been paused. Please update WooCommerce to the latest version to continue enjoying WooCommerce Blocks.', 'woo-gutenberg-products-block' ); ?></p>
							</div>
							<?php
						}
					}
				);
			}
		}
		$end_time      = hrtime( true );
		$time_executed = 'Time spent:' . ( $end_time - $start_time );
		var_dump( $time_executed );
		return $has_needed_dependencies;
	}
```

The _most_ time I saw reported on multiple refreshes across different pages (including frontend/backend) was 4565661 nano seconds (or 0.004565661 seconds). IMO that is extremely negligible and in keeping with the performance impact of including PHP files.